go/oasis-node/cmd/debug/txsource/workload/queries.go

```diff
@@ -874,13 +876,15 @@ func (q *queries) Run(
 
 		// Ensure the node appears synced before doing queries. Given that nodes can be killed and
 		// held stopped for a while, they need to sync once they come back.
-		if isSynced, _ := q.control.IsSynced(loopCtx); !isSynced {
-			_ = q.control.WaitSync(loopCtx)
-			time.Sleep(1 * time.Second)
-			continue
+		switch isSynced, _ := q.control.IsSynced(loopCtx); isSynced {
+		case true:
+			// Node is synced, proceed with queries.
+			err = q.doQueries(loopCtx, rng)
+		case false:
+			// Node is not synced, wait for it to sync (at most for queriesIterationTimeout).
+			err = q.control.WaitSync(loopCtx)
```

Comment:
```suggestion
			err = q.control.WaitSync(loopCtx)
			q.iteration--
```

Reply:
If context expires (or any other error occurs), the error will be handled below and the function will return.

Reply:
Yes this is on purpose as the node should not keep syncing forever.

Reply:
I thought about tweaking the iteration count and thought that it shouldn't matter that much, but sure it is better hygiene :)

Reply:
I changed it around so the iteration counter is only updated in the normal case.